fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java

```diff
@@ -280,7 +282,25 @@ public static Map<String, String> toHudiProperties(Table metastoreTable,
 
             TypeInfo fieldInfo = getTypeInfoFromTypeString(field.get().getType());
             columnNamesBuilder.append(field.get().getName());
-            columnTypesBuilder.append(fieldInfo.getTypeName());
+
+            String type = fieldInfo.getTypeName();
+            if (type.equals("timestamp")) {
+                Schema fieldSchema = hudiField.schema();
+                List<Schema> nonNullMembers = fieldSchema.getTypes().stream()
```

Comment:
What does this piece of code mean?

Reply:
> Because parquet avro's get type returns a union representation, such as int, the return is [null, int], with a traversal to remove the null to get the true type